### PR TITLE
Add stack-hci-vm

### PR DESCRIPTION
**Pull request by wyatt-troia-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
  - No, because I couldn't figure out how to get azdev to recognize my extension
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).
  - The linked doc doesn't explain where the extension description/summary is defined, so no.

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by ggaurav10 — CHANGES_REQUESTED** [review]
some cli flags which shouldn't be exposed are not suppressed.

File: src/index.json [hunk 1]
```diff
@@ -56400,6 +56400,51 @@
                 "sha256Digest": "58074ff73a72a1a9c76d406a4b6988a7b714dfaa9259b92172a7b6f66680db31"
             }
         ],
+        "stack-hci-vm": [
+            {
+                "downloadUrl": "https://hybridaksstorage.z13.web.core.windows.net/SelfServiceVM/CLI/stack_hci_vm-0.1.2-py3-none-any.whl",
+                "filename": "stack_hci_vm-0.1.2-py3-none-any.whl",
+                "metadata": {
+                    "azext.isExperimental": true,
```
Comment (kairu-ms) [inline src/index.json:56408]: The isExperimental is not used any more. Please use isPreview instead

File: src/index.json [hunk 2]
```diff
@@ -56400,6 +56400,51 @@
...
+                    "azext.minCliCoreVersion": "2.15.0",
+                    "classifiers": [
+                        "Development Status :: 4 - Beta",
+                        "Intended Audience :: Developers",
+                        "Intended Audience :: System Administrators",
+                        "Programming Language :: Python",
+                        "Programming Language :: Python :: 3",
+                        "Programming Language :: Python :: 3.6",
+                        "Programming Language :: Python :: 3.7",
+                        "Programming Language :: Python :: 3.8",
```
Comment (kairu-ms) [inline src/index.json:56418]: BTW the python version which less than 3.8 is not supported. Please update it as well

File: src/index.json [hunk 3]
```diff
@@ -56400,6 +56400,51 @@
...
+                                    "name": "Microsoft Corporation",
+                                    "role": "author"
+                                }
+                            ],
+                            "document_names": {
+                                "description": "DESCRIPTION.rst"
+                            },
+                            "project_urls": {
+                                "Home": "https://github.com/Azure/azure-cli-extensions/tree/master/src/stack-hci-vm"
+                            }
```
Comment (kairu-ms) [inline src/index.json:56435]: As the source code of this extension is not in Azure/azure-cli-extensions repo, please correct the descriptions here.

Comment (wyatt-troia-msft) [reply src/index.json]: What all do I need to correct?

File: src/index.json (hunk 2 above)
Comment (wyatt-troia-msft) [reply src/index.json]: Should I keep the line without any version of Python specified?

File: src/index.json (hunk 1 above)
Comment (wyatt-troia-msft) [reply src/index.json:56408]: Done

**Review by ggaurav10 — COMMENTED** [review]
looks good from stack-hci-vm code perspective.

**Review by ggaurav10 — APPROVED** [review]

File: src/index.json (hunk 3 above)
Comment (Arkanayan) [reply src/index.json]: Is it okay if we remove the whole section?
@kairu-ms

Comment (kairu-ms) [reply src/index.json]: These metadata are retrieved from the extension source code. So you should correct them in the extension code.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**